### PR TITLE
add channel arguments to snaps in the action

### LINKED ISSUE #27: Allow for setting snap channels of all auto-installed snaps
`actions-operator` conveniently [installs charmcraft](https://github.com/charmed-kubernetes/actions-operator/blob/a32a699/src/bootstrap/index.ts#L85). However, I would like to install `charmcraft` from `latest/candidate`, since that's the channel pointing at `1.3.1`. I can run `sudo snap refresh charmcraft --channel latest/candidate` after the `actions-operator` step, but it would be nice if I could just add a `charmcraft_channel: latest/candidate` section to the `with:` block.

This also seems like something that should be done for all snaps installed by `actions-operator`, since various use cases like this will probably crop up for other snaps as well.

### PR DESCRIPTION
**Pull request by ca-scribner**
resolves #27

### CODE REVIEW

**Comment by ca-scribner** [comment]
I see the other channel args were left as `default_value=null` and then use if statements.  I ended up using `default=latest/stable` so all calls could be `snap install ... --channel=${this_channel}` to avoid a few extra lines, but I can change it to match the current format if that's preferred.

File: src/bootstrap/index.ts [hunk 1]
```diff
@@ -81,9 +84,9 @@ async function run() {
         core.endGroup();
         core.startGroup("Install tools");
         await exec.exec("sudo snap install jq");
-        await exec.exec("sudo snap install charm --classic");
-        await exec.exec("sudo snap install charmcraft --classic");
-        await exec.exec("sudo snap install juju-bundle --classic");
+        await exec.exec("sudo snap install charm --classic --channel=${charm_channel}");
+        await exec.exec("sudo snap install charmcraft --channel=${charmcraft_channel}");
+        await exec.exec("sudo snap install juju-bundle --channel=${juju_channel}");
```
Comment (knkski) [inline src/bootstrap/index.ts:89]: This is setting it for `juju-bundle` instead of for `juju`

**Review by knkski — APPROVED** [review]

File: src/bootstrap/index.ts [hunk 2]
```diff
@@ -77,13 +81,13 @@ async function run() {
         await exec.exec("sudo --preserve-env=http_proxy,https_proxy,no_proxy pip3 install tox");
         core.endGroup();
         core.startGroup("Install Juju");
-        await exec.exec("sudo snap install juju --classic");
+        await exec.exec("sudo snap install juju --classic --channel=${juju_channel}");
```
Comment (johnsca) [inline src/bootstrap/index.ts:84]: You have to use backtick quotes for interpolation to work (this also applies to the other lines).

```suggestion
        await exec.exec(`sudo snap install juju --classic --channel=${juju_channel}`);
```

**Review by johnsca — CHANGES_REQUESTED** [review]
In addition to the comment below, you also need to run `npm run build`. There's a pre-commit hook in the repo under `.github/hooks/pre-commit` that I recommend you symlink into your `.git/hooks/` directory to catch that.

Also, the PR probably doesn't need to be a draft anymore.

File: src/bootstrap/index.ts [hunk 3]
```diff
@@ -77,13 +81,13 @@ async function run() {
...
         core.endGroup();
         core.startGroup("Install tools");
         await exec.exec("sudo snap install jq");
-        await exec.exec("sudo snap install charm --classic");
-        await exec.exec("sudo snap install charmcraft --classic");
-        await exec.exec("sudo snap install juju-bundle --classic");
+        await exec.exec("sudo snap install charm --classic --channel=${charm_channel}");
```
Comment (ca-scribner) [inline src/bootstrap/index.ts:88]:
```suggestion
        await exec.exec(`sudo snap install charm --classic --channel=${charm_channel}`);
```

File: src/bootstrap/index.ts [hunk 4]
```diff
@@ -77,13 +81,13 @@ async function run() {
...
+        await exec.exec("sudo snap install charmcraft --channel=${charmcraft_channel}");
+        await exec.exec("sudo snap install juju-bundle --channel=${juju_bundle_channel}");
```
Comment (ca-scribner) [inline src/bootstrap/index.ts:90]:
```suggestion
        await exec.exec(`sudo snap install charm --classic --channel=${charm_channel}`);
        await exec.exec(`sudo snap install charmcraft --channel=${charmcraft_channel}`);
        await exec.exec(`sudo snap install juju-bundle --channel=${juju_bundle_channel}`);
```

**Review by johnsca — APPROVED** [review]

**Pull request merged**